experimental/Origami/src/deck_group.jl

```diff
@@ -0,0 +1,54 @@
+function in_deck_group(o::Origami, sigma::PermGroupElem)
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    for i in 1:degree(o)
+        if (i^sigma)^h != (i^h)^sigma || (i^sigma)^v != (i^v)^sigma
+            return false
+        end
+    end
+    return true
+end
+
+function deck_group(o::Origami)
+    function candidate_for_deck(origami, j)
+        sheets_to_visit = 2:degree(origami)
+        sigma = [j]
+        append!(sigma, fill(-1, degree(origami) - 1))
+        while length(sheets_to_visit) > 0
+            found_predecessor = false
+            h = horizontal_perm(origami)
+            v = vertical_perm(origami)
+            for i in sheets_to_visit
+                for tau in [h, v]
+                    pos_iterator = searchsorted(sheets_to_visit, i^(tau^-1))
+                    if first(pos_iterator) > last(pos_iterator)
+                        found_predecessor = true
+                        sigma[i] = sigma[i^(tau^-1)]^tau
+                        deleteat!(sheets_to_visit, findfirst(item -> item == i,
+                                                            sheets_to_visit))
+                        break;
+                    end
+                end
+                if found_predecessor
+                    break;
+                end
```

Comment:
```suggestion
                if found_predecessor
                    break
                end
```
or, more idiomatically:
```suggestion
                found_predecessor && break
```